File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/home/HomeView.java
package io.muun.apollo.presentation.ui.home;

import io.muun.apollo.presentation.ui.base.BaseView;

public interface HomeView extends BaseView {

    /**
     * Takes user to SecurityCenter screen.
     */
    void navigateToSecurityCenter();

    /**
     * Takes user to SecurityCenter screen.
     */
    void showWelcomeToMuunDialog();

    /**
     * Show Taproot celebration.
     */
    void showTaprootCelebration();
}

File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/fragments/home/HomeView.kt
package io.muun.apollo.presentation.ui.fragments.home

import io.muun.apollo.domain.model.CurrencyDisplayMode
import io.muun.apollo.domain.model.Operation
import io.muun.apollo.presentation.ui.base.BaseView

interface HomeView : BaseView {

    fun setState(homeState: HomePresenter.HomeState)

    fun setNewOp(newOp: Operation, mode: CurrencyDisplayMode)

    fun showTooltip()
}